name.abuchen.portfolio/src/name/abuchen/portfolio/util/webaccess/WebAccess.java

```diff
@@ -0,0 +1,197 @@
+
+    public static WebAccessBuilder builder()
+    {
+        return new WebAccess.WebAccessBuilder();
+    }
+
+    private WebAccess()
+    {
+    }
+
+    public String getScheme()
+    {
+        return scheme;
+    }
+
+    public void setScheme(String scheme)
+    {
+        this.scheme = scheme;
+    }
+
+    public String getHost()
+    {
+        return host;
+    }
+
+    public void setHost(String host)
+    {
+        this.host = host;
+    }
+
+    public String getPath()
+    {
+        return path;
+    }
+
+    public void setPath(String path)
+    {
+        this.path = path;
+    }
+
+    public List<WebAccessHeader> getHeaders()
+    {
+        return headers;
+    }
+
+    public void setHeaders(List<WebAccessHeader> headers)
+    {
+        this.headers = headers;
+    }
+
+    public List<WebAccessParameter> getParameters()
+    {
+        return parameters;
+    }
+
+    public void setParameters(List<WebAccessParameter> parameters)
+    {
+        this.parameters = parameters;
+    }
+
+    public CloseableHttpResponse getResponse()
+    {
+        return response;
+    }
+
+    public void setResponse(CloseableHttpResponse response)
+    {
+        this.response = response;
+    }
+
+    public String getDocument()
+    {
+
+        List<Header> lstHeaders = new ArrayList<>();
+        if (headers != null)
+        {
+            for (WebAccessHeader header : headers)
+            {
+                lstHeaders.add(new BasicHeader(header.getParam(), header.getValue()));
+            }
+        }
+
+        CloseableHttpClient client = HttpClientBuilder.create().setDefaultRequestConfig(defaultRequestConfig)
+                        .setDefaultHeaders(lstHeaders)
+                        .setUserAgent("Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.132 Safari/537.36") //$NON-NLS-1$
+                        .build();
+
+        URIBuilder uriBuilder = new URIBuilder().setScheme(scheme).setHost(host).setPath(path);
+
+        List<NameValuePair> lstParameters = new ArrayList<NameValuePair>();
+        if (parameters != null)
+        {
+            for (WebAccessParameter parameter : parameters)
+            {
+                lstParameters.add(new BasicNameValuePair(parameter.getParam(), parameter.getValue()));
+            }
+        }
+        uriBuilder.setParameters(lstParameters);
+
+        URL objectURL;
+        try
+        {
+            objectURL = uriBuilder.build().toURL();
+            CloseableHttpResponse response = client.execute(new HttpGet(objectURL.toString()));
+            setResponse(response);
+            if (response.getStatusLine().getStatusCode() != HttpStatus.SC_OK)
+                throw new IOException(objectURL.toString() + " --> " + response.getStatusLine().getStatusCode()); //$NON-NLS-1$
+
+            this.document = EntityUtils.toString(response.getEntity());
+        }
+        catch (URISyntaxException | IOException e)
```

Comment:
Die ```IOException``` muss auf jeden Fall in die Signatur der Methode und darf hier nicht abgefangen werden. Ansonsten kann PP dem Benutzer keine Fehlermeldungen zurückgeben.

Reply:
Ich hoffe das es im Try # 9 so passt...